### PR TITLE
[SPARK-38561][K8S][DOCS][3.3] Add doc for `Customized Kubernetes Schedulers`

### PR DESCRIPTION
**Pull request by Yikun**
### What changes were proposed in this pull request?
This is PR to doc for basic framework capability for Customized Kubernetes Schedulers.

### Why are the changes needed?
Guide user how to use spark on kubernetes custom scheduler

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
CI passed

### CODE REVIEW

File: docs/running-on-kubernetes.md [hunk 1]
```diff
@@ -1713,6 +1713,25 @@ spec:
     image: will-be-overwritten
 ```
 
+#### Customized Kubernetes Schedulers for Spark on Kubernetes
+
+Spark allows users to specify a custom Kubernetes schedulers.
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1718]: `a custom Kubernetes schedulers` -> `custom Kubernetes schedulers`

File: docs/running-on-kubernetes.md [hunk 2]
```diff
@@ -1713,6 +1713,25 @@ spec:
...
+
+1. Specify scheduler name.
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1720]: `Specify` -> `Specify a`

File: docs/running-on-kubernetes.md [hunk 3]
```diff
@@ -1713,6 +1713,25 @@ spec:
...
+
+   Users can specify a custom scheduler using <code>spark.kubernetes.scheduler.name</code> or
+   <code>spark.kubernetes.{driver/executor}.scheduler.name</code> configuration.
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1723]: `{driver/executor}` -> `{driver|executor}`.

File: docs/running-on-kubernetes.md [hunk 4]
```diff
@@ -1713,6 +1713,25 @@ spec:
...
+
+2. Specify scheduler related configurations.
+
+   To configure the custom scheduler the user can use [Pod templates](#pod-template), add labels (<code>spark.kubernetes.{driver,executor}.label.*</code>)  and/or annotations (<code>spark.kubernetes.{driver/executor}.annotation.*</code>).
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1727]: Hmm. It seems that you missed the most important thing, `PodGroupTemplate`, here.

File: docs/running-on-kubernetes.md [hunk 5]
```diff
@@ -1713,6 +1713,25 @@ spec:
...
+
+3. Specify scheduler feature step.
+
+   Users may also consider to use <code>spark.kubernetes.{driver/executor}.pod.featureSteps</code> to support more complex requirements, including but not limited to:
+  - Create additional Kubernetes custom resources for driver/executor scheduling.
+  - Set scheduler hints according to configuration or existing Pod info dynamically.
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1733]: What does this refer in Volcano scheduler?

Comment (Yikun) [reply docs/running-on-kubernetes.md:1733]: > Set scheduler hints according to configuration

create pg (contains scheduler hints) according to `PodGroupTemplate`

> existing Pod info dynamically.

such as we can set priority according driver.priority, of course, it didn't use it in volcano current implementations

File: docs/running-on-kubernetes.md (hunk 4 above)
Comment (Yikun) [reply docs/running-on-kubernetes.md:1727]: I guess this has been included by  doc of feature step section? https://github.com/apache/spark/pull/35955#discussion_r833873923

**Comment by Yikun** [comment]
@dongjoon-hyun @holdenk Would you mind take a look again on this? the doc for Customized Kubernetes Schedulers for 3.3 branch.

**Comment by Yikun** [comment]
Address comments, also sync changes on master: https://github.com/apache/spark/pull/36001

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by Yikun** [comment]
@dongjoon-hyun Thanks!

**Pull request closed without merging**